### PR TITLE
JAVA-81 - TransactionService

### PR DESCRIPTION
**Pull request by fboucquez**
Hi guys, this is my implementation of the transaction service. A service that helps users announce transactions given them feedback when the transactions have been completed or failed. 

I've migrated all the integration tests to use the new service. They seem to be fine. 
But, I've ported the `CreatingAnEscrowContractWithAggregateBondedTransactionIntegrationTest` but it's timing out for some reason. I do have tests using the hash/bonded composed transaction of `TransactionService.announceHashLockAggregateBonded`. 

@dgarcia360, were you able to run your example?

https://github.com/nemtech/nem2-docs/blob/master/source/resources/examples/typescript/aggregate/CreatingAnEscrowContractWithAggregateBondedTransaction.ts#L75-L115

### CODE REVIEW

File: sdk-core/src/main/java/io/nem/sdk/model/transaction/SignedTransaction.java [hunk 1]
```diff
@@ -24,16 +26,28 @@
  */
 public class SignedTransaction {
 
+    private final Address signer;
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/model/transaction/SignedTransaction.java]: The SignedTransaction Object doesn't have the public account.

https://github.com/nemtech/nem2-sdk-java/blob/a28a088dd91496db4e69266aaf0358b76a636ea1/sdk-core/src/main/java/io/nem/sdk/model/transaction/SignedTransaction.java

Having the address helps us tracking who signed the SignedTransaction, especially when you want to get status/completed message back via the listener.

https://github.com/nemtech/nem2-sdk-java/blob/a28a088dd91496db4e69266aaf0358b76a636ea1/sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java#L49 

If not, you end up sending 2 parametres per transaction, the signer address/account, and the hash/SignedTransaction. This gets complicated in methods like `announceHashLockAggregateBonded`

https://github.com/nemtech/nem2-sdk-java/blob/a28a088dd91496db4e69266aaf0358b76a636ea1/sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java#L66

The alternative is to change the SignedTransaction's signer from an Address to a PublicAccount. I want to keep the object minimal. PublicAccount holds more information.

File: sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import io.reactivex.Observable;
+import org.apache.commons.lang3.Validate;
+
+/**
+ * Implementation of {@link TransactionService}.
+ */
+public class TransactionServiceImpl implements TransactionService {
+
+    private final TransactionRepository transactionRepository;
+
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java]: The main documentation is in the interface.

https://github.com/nemtech/nem2-sdk-java/blob/a28a088dd91496db4e69266aaf0358b76a636ea1/sdk-core/src/main/java/io/nem/sdk/api/TransactionService.java

The implementation is basically a copy of the public interface methods or some implementation details. I'll add some docs to the private methods though

File: sdk-core/src/main/java/io/nem/sdk/model/transaction/SignedTransaction.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rg911) [reply sdk-core/src/main/java/io/nem/sdk/model/transaction/SignedTransaction.java]: PublicAccount from publicKey. Address param is redundant.

[the remark this answers is not in the document]
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/model/transaction/SignedTransaction.java]: So you prefer signer to be PublicAccount?

[the remark this answers is not in the document]
Comment (rg911) [reply sdk-core/src/main/java/io/nem/sdk/model/transaction/SignedTransaction.java]: what I mean is you can get address off the publicAccount

**Pull request merged**